java/connector-node/risingwave-sink-mock-flink/risingwave-sink-mock-flink-runtime/src/main/java/com/risingwave/connector/sinkwriter/SinkWriterV2Impl.java

```diff
@@ -0,0 +1,160 @@
+        }
+
+        try {
+            if (this.sink instanceof TwoPhaseCommittingSink) {
+                this.committer = ((TwoPhaseCommittingSink<?, CommT>) this.sink).createCommitter();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void writeRow(RowData rowData) {
+        try {
+            if (this.writer == null) {
+                this.writer = sink.createWriter(new SinkWriterContextV2());
+            }
+            this.writer.write(
+                    rowData,
+                    new SinkWriter.Context() {
+                        @Override
+                        public long currentWatermark() {
+                            return 0;
+                        }
+
+                        @Override
+                        public Long timestamp() {
+                            return null;
+                        }
+                    });
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void write(Iterator<SinkRow> rows) {
+        while (rows.hasNext()) {
+            SinkRow row = rows.next();
+            writeRow(new RowDataImpl(row, getTableSchema().getNumColumns()));
+        }
+    }
+
+    @Override
+    public void sync() {
+        try {
+            if (isAsyncSink) {
+                if (writer == null) {
+                    return;
+                }
+                this.writer.flush(true);
+            } else {
+                this.writer.flush(false);
+                if (committer != null) {
+                    Collection<CommT> objects =
+                            ((TwoPhaseCommittingSink.PrecommittingSinkWriter<?, CommT>) this.writer)
+                                    .prepareCommit();
+                    if (!objects.isEmpty()) {
+                        List<Committer.CommitRequest<CommT>> collect =
+                                objects.stream()
+                                        .map(
+                                                (a) -> {
+                                                    return (Committer.CommitRequest<CommT>)
+                                                            new CommitRequestImpl<>(a);
+                                                })
+                                        .collect(Collectors.toList());
+                        committer.commit(collect);
+                    }
+                }
+                if (this.writer instanceof TwoPhaseCommittingSink.PrecommittingSinkWriter) {
+                    ((TwoPhaseCommittingSink.PrecommittingSinkWriter<?, ?>) this.writer)
+                            .prepareCommit();
+                }
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        } finally {
```

Comment:
From my understanding we don't need the `finally` clause and the logic of recreating the writer for async writer can be moved to the `try` bracket, because when exception happens, our framework should ensure that `drop` is called, and we don't need to handle the state cleaning here.

Besides, the logic of `this.writer = null` can be moved to `closeWriter`.